src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -28,20 +27,17 @@
 class DelegatingLoader extends BaseDelegatingLoader
 {
     protected $parser;
-    protected $logger;
     private $loading = false;
 
     /**
      * Constructor.
      *
      * @param ControllerNameParser    $parser   A ControllerNameParser instance
-     * @param LoggerInterface         $logger   A LoggerInterface instance
      * @param LoaderResolverInterface $resolver A LoaderResolverInterface instance
      */
-    public function __construct(ControllerNameParser $parser, LoggerInterface $logger = null, LoaderResolverInterface $resolver)
+    public function __construct(ControllerNameParser $parser, LoaderResolverInterface $resolver)
```

Comment:
So, what should be done in order to have both old and new api compatible in 2.8 ? Something like:
```php
/**
 * @param ControllerNameParser                    $parser   A ControllerNameParser instance
 * @param LoggerInterface|LoaderResolverInterface $logger   A LoggerInterface instance
 * @param LoaderResolverInterface|null            $resolver A LoaderResolverInterface instance
 */
public function __construct(ControllerNameParser $parser, $logger = null, $resolver = null)
{
    $this->parser = $parser;
    $this->logger = $logger instanceof LoggerInterface ? $logger : null;

    parent::__construct($logger instanceof LoaderResolverInterface ? $logger : $resolver);
}
```
(the old api will still be used by the FrameworkBundle) and raising a deprecation warning when subclasses access the `$logger` property by using a magic getter  as @Tobion suggested ? 
Pretty confusing.